tests/sbatch_test.py

```diff
@@ -13,13 +13,66 @@
 # EOF
 
 
-def test_srw_data_file(new_user_fc):
+def test_srw_cancel(fc):
+    from pykern.pkcollections import PKDict
+    from pykern.pkdebug import pkdp
+    from pykern.pkunit import pkeq, pkfail, pkok
+    import subprocess
+    import time
+
+    def _squeue_num_jobs():
+        # slurm is a shared resource (not unique to each test) so this
+        # has the potential to conflict with jobs users have started
+        # outside of tests.
+        # Will be fixed by https://github.com/radiasoft/sirepo/issues/7012
+        # -1 to remove header line
+        return int(subprocess.check_output("squeue | wc -l", shell=True)) - 1
+
+    pkeq(0, _squeue_num_jobs())
+    sim_name = "Young's Double Slit Experiment"
+    compute_model = "multiElectronAnimation"
+    fc.sr_sbatch_login(compute_model, sim_name)
+    d = fc.sr_sim_data(sim_name=sim_name)
+    d.models[compute_model].jobRunMode = fc.sr_job_run_mode
+    r = fc.sr_post(
+        "runSimulation",
+        PKDict(
+            models=d.models,
+            report=compute_model,
+            simulationId=d.models.simulation.simulationId,
+            simulationType=d.simulationType,
+        ),
+        expect_completed=False,
+    )
+
+    for _ in range(5):
+        pkok(
+            r.state in ("running", "pending"),
+            "runSimulation did not start: reply={}",
+            r,
+        )
+        if r.state == "running" and _squeue_num_jobs() == 1:
+            break
+        time.sleep(5)
+        r = fc.sr_post("runStatus", r.nextRequest)
+    else:
+        pkfail("failed to start running in sirepo and slurm reply={}", r)
+    r = fc.sr_post("runCancel", r.nextRequest)
+    for _ in range(5):
+        if _squeue_num_jobs() == 0:
+            break
+        time.sleep(5)
```

Comment:
I think we should modularize this with a name:

```py
while self.iter_sleep("slurm", "runCancel"):
    do some check

def iter_sleep(self, kind, op_desc):
    def _setup():
        rv = PKDict(_ITER_SLEEP[kind])
        rv.countdown = range(rv.count, -1, -1)
        return rv
        
    s = _setup()
    for i  in s.countdown():
        yield
        if i > 0:
            time.sleep(s.sleep_secs)
    else:
        pkfail("timeout secs={} kind={} op_desc={}", s.total_secs, kind, op_desc)
```

Reply:
5ce403daf